libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5259,6 +5265,182 @@ void GCS_MAVLINK::initialise_message_intervals_for_stream(GCS_MAVLINK::streams i
     }
 }
 
+#if HAL_MAVLINK_INTERVALS_FROM_FILES_ENABLED
+// open and read contents of path, setting message intervals from each
+// line
+DefaultIntervalsFromFiles::DefaultIntervalsFromFiles(uint16_t max_num)
+{
+    if (max_num == 0) {
+        return;
+    }
+    _intervals = (struct from_file_default_interval*)malloc(max_num * sizeof(struct from_file_default_interval));
```

Comment:
array new would be nice, but not required

Reply:
Done